core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java

```diff
@@ -4262,6 +4262,87 @@ public void testLatestByAllValueIndexedColumn() throws Exception {
+    }
+
+    @Test
+    public void testLatestByPartiallyIndexedFilteredSymbolIn() throws Exception {
+        testLatestByFilteredBySymbolIn("create table x (\n" +
+                "  ts timestamp,\n" +
+                "  node symbol index,\n" +
+                "  metric symbol,\n" +
+                "  value long) \n" +
+                "  timestamp(ts) partition by day");
+    }
+
+    private void testLatestByFilteredBySymbolIn(String ddl) throws Exception {
+        assertMemoryLeak(() -> {
+            compiler.compile(ddl, sqlExecutionContext);
+
+            executeInsert("insert into x values ('2021-11-17T17:35:01.000000Z', 'node1', 'cpu', 100)");
+            executeInsert("insert into x values ('2021-11-17T17:35:01.000000Z', 'node2', 'cpu', 75)");
+            executeInsert("insert into x values ('2021-11-17T17:35:01.000000Z', 'node3', 'cpu', 25)");
+            executeInsert("insert into x values ('2021-11-17T17:35:02.000000Z', 'node1', 'memory', 2000)");
+            executeInsert("insert into x values ('2021-11-17T17:35:02.000000Z', 'node2', 'memory', 3000)");
+            executeInsert("insert into x values ('2021-11-17T17:35:02.000000Z', 'node3', 'memory', 4000)");
+
+            TestUtils.assertSql(
+                    compiler,
+                    sqlExecutionContext,
+                    "select metric, sum(value) from x latest by node \n" +
+                            "where node in ('node1', 'node2') and metric in ('cpu')",
+                    sink,
+                    "metric\tsum\n" +
+                            "cpu\t175\n"
+            );
+        });
+    }
+
+    @Test
+    public void testLatestBySelectAllFilteredBySymbolIn() throws Exception {
+        assertMemoryLeak(() -> {
+            compiler.compile("create table x (\n" +
+                    "  ts timestamp,\n" +
+                    "  node symbol,\n" +
+                    "  metric symbol,\n" +
+                    "  value long) \n" +
+                    "  timestamp(ts) partition by day", sqlExecutionContext);
+
+            executeInsert("insert into x values ('2021-11-17T17:35:01.000000Z', 'node1', 'cpu', 100)");
+            executeInsert("insert into x values ('2021-11-17T17:35:01.000000Z', 'node2', 'cpu', 75)");
+            executeInsert("insert into x values ('2021-11-17T17:35:02.000000Z', 'node1', 'memory', 2000)");
+            executeInsert("insert into x values ('2021-11-17T17:35:02.000000Z', 'node2', 'memory', 3000)");
+
+            TestUtils.assertSql(
+                    compiler,
+                    sqlExecutionContext,
+                    "select * from x latest by node, metric \n" +
+                            "where node in ('node2') and metric in ('cpu', 'memory')",
+                    sink,
+                    "ts\tnode\tmetric\tvalue\n" +
+                            "2021-11-17T17:35:01.000000Z\tnode2\tcpu\t75\n" +
+                            "2021-11-17T17:35:02.000000Z\tnode2\tmemory\t3000\n"
+            );
```

Comment:
can you add indexes on symbols one by one and repeat the assertion please?

Reply:
Done in 265fd52e4de65e3d66e076a791ea44c0452b272b